TinyMCE.Blazor/Editor.razor

```diff
@@ -18,186 +20,307 @@
     {
-      streamMap[streamId] = acc;
-    }
-  }
-
-  protected async Task PushValue(string nextVal)
-  {
-    int chunks = (nextVal.Length / _chunkSize) + 1;
-    string streamId = String.Format("{0:d9}", (DateTime.Now.Ticks / 10) % 1000000000);
-    for (int i = 0; i < chunks; i++) {
-      string chunk = nextVal.Substring( _chunkSize * i, nextVal.Length < (_chunkSize * ( i + 1)) ? (nextVal.Length - (_chunkSize * i)) : _chunkSize);
-      await JSRuntime.InvokeVoidAsync("tinymceBlazorWrapper.updateValue", Id, streamId, chunk, i + 1, chunks);
-    }
-  }
-
-  protected async Task ChangeMode(bool disabled)
-  {
-    await JSRuntime.InvokeVoidAsync("tinymceBlazorWrapper.updateMode", Id, disabled);
-  }
-    
-  protected override async Task OnAfterRenderAsync(bool firstRender)
-  {
-    if (firstRender)
-    {
-      blazorConf = new Dictionary<string, object>();
-      blazorConf.Add("inline", Inline);
-      string tinySrc = ScriptSrc != null ? ScriptSrc : $"https://cdn.tiny.cloud/1/{ApiKey}/tinymce/{CloudChannel}/tinymce.min.js";
-      blazorConf.Add("src", tinySrc);
-      blazorConf.Add("jsConf", JsConfSrc);
-      blazorConf.Add("modelEvents", ModelEvents);
-      Dictionary<string, object> merged = Merge(Conf, AdditionalAttributes);
-      blazorConf.Add("conf", merged);
-      if (!_setDisable && merged.ContainsKey("readonly"))
-      {
-        _disable = merged["readonly"] is bool ? (bool)merged["readonly"] : false;
-      }
-      blazorConf.Add("disabled", Disable);
-      // TBD: Load the events
-        @*foreach (KeyValuePair<string, object> kvp in AdditionalAttributes)
+        return Task.FromResult(Value);
+    }
+
+    [JSInvokable("UpdateModel")]
+    public void OnChangeContent(string streamId, int index, string val, int size)
+    {
+        PushUpdate(streamId, index, val, size, (content) =>
+        {
+            _value = content;
+            ValueChanged.InvokeAsync(content);
+        });
+    }
+
+    [JSInvokable("UpdateText")]
+    public void OnChangeText(string streamId, int index, string val, int size)
+    {
+        PushUpdate(streamId, index, val, size, (content) =>
+        {
+            _text = content;
+            TextChanged.InvokeAsync(content);
+        });
+    }
+
+    /*
+     * By default, hooked up with TinyMce's onchange event;
+     */
+
+    [JSInvokable("OnChange")]
+    public void OnChange()
+    {
+        if (!ValidateOnInput)
+        {
+            NotifyFieldChanged();
+        }
+    }
+
+    /*
+    * By default, hooked up with TinyMce's oninput event;
+    */
+
+    [JSInvokable("OnInput")]
+    public void OnInput()
+    {
+        if (ValidateOnInput)
+        {
+            NotifyFieldChanged();
+        }
+    }
+
+    private void NotifyFieldChanged()
+    {
+        if (For == null)
+        {
+            return;
+        }
+    // Notify field changed to trigger validation
+        EditContext?.NotifyFieldChanged(FieldIdentifier.Create(For));
```

Comment:
I believe that if we inherit from `InputBase` we can just call the `FieldIdentifier` property

Reply:
InputBase comes with a few bagages. So far, if the Editor is not used inside an EditForm, it would behave exactly like before. The `change` and `input` events will be triggered and `OnChange` and `OnInput` will be called, but they do nothing. The question is, do we want to inherit from InputBase?

Reply:
Inherit from InputBase has its benefits. However, from reading the source code `InputBase`, I believe it requires the component to be used inside an EditForm.